BWWalkthrough/BWWalkthroughViewController.swift

```diff
@@ -252,28 +189,72 @@ At the moment it's only used to perform custom animations on didScroll.
         
         // Hide/Show navigation buttons
         
-        if currentPage == controllers.count - 1{
+        updateButtons()
+    }
+    
+    /**
+    Updates previous and next buttons.
+    */
+    private func updateButtons() {
+        
+        if currentPage == controllers.count - 1 {
             nextButton?.hidden = true
-        }else{
+        } else {
             nextButton?.hidden = false
         }
         
-        if currentPage == 0{
+        if currentPage == 0 {
             prevButton?.hidden = true
-        }else{
+        } else {
             prevButton?.hidden = false
         }
+        
+        if let currentPage = controllers[currentPage] as? BWWalkthroughPage {
+            switch currentPage.pageControlPreference {
+            case .Default:
+                closeButton?.hidden = false
+            case .HideCloseButton:
```

Comment:
It's future proofing. There may be further 'page control preferences' outside of just the close button.